### DISCUSSION TITLE
8.1 멘토링

### DISCUSSION THREAD

**Discussion by t4e1** [3sam5oh/webtoon-recommendation-service#108] [멘토링 일지]
### Terrafom 관련 피드백
- 실제 EC2 등 만들어 본 다음에 통신 되는지 확인해보기
-  EKS 상관 없이 ASG, ubuntu, user-data 통해서 컨테이너 서빙이 가능한 상태로 부팅이 되도록 만들기
- keychain을 만들고 security group을 asg에게 주고 ssh로 접속 할 수 있도록
- Instance profile을 해당 ASG의 LT에 마찬가지로 같이 넣어주기
- LT(Launch Template) : AMI  
- Instance profile = IAM role wrapper
- terrafom 공부 할 때, 익숙하지 않을 때는 console 에서 먼저 손으로 만들어보고, 별도로 code로 인프라를 만든 다음에 terrrafom state import 등으로 기존 인프라를 import 한 다음에 코드를 plan으로 동기화 시키면서 테스트해보기

### 그라파나 대시보드 
- 서비스 별로 목적에 따라 별도의 대시보드 생성 
- nodeexporter 는 기본적으로 봐야한다. 

### TLS 관련 가이드 
- HPBN 확인

### k6
- 스크립트를 테스트별로 따로 만들어야 할지? 
- k6 위치 
  - 일반적으로 별도의 pod 으로 돌리는게 일반적 
- 중요한 점은 테스트도 완성되면 시스템으로 만들 수 있어야 함

### 시크릿 관리 
- sops 랑 secret manager 사용 
- 직접 코드를 관리해야 하는 경우에는 sops를 사용 

### 프로젝트 진행 상황 
- 클라우드 상에서 싱크를 맞추기에는 힘들 것 같음 
- 싱크를 맞추려면 k8s 에서 모든 걸 올리고 공유하는 게 우선일 것 같다. 
- 네트워크 이슈나 보안 관련은 미리 겪어 볼 수 있게 하는게 좋음  

### Trace
- Java OpenTelemetry 라이브러리 사용 
- 중요한 function 이 호출될 때마다 Endpoint 로 보낸다.
- Grafana tempo 등 설치해서 그쪽으로 트레이스를 보내고 Grafana 에서 모니터링 확인 
- 코드 쪽을 건드는 부분( Java 랭귀지 관련해서 관심 있는 사람이 하는 게 좋음) 

## 2. 별도
- 웹 드라이버 프로토콜
- CDP (Chrome Devtools Protocol)
- 사용해보면 웹 개발자 입장에서 도움이 될 수 있음 
- ex: 셀레니움, playwright, go-rod등...
- selenium 이 웹 드라이버 프로토콜을 사용 (크로니움 기반이 아니어도 사용 가능)
- CDP 가 좀 더 많은 기능을 제공해줌

**Comment by t4e1** [comment]
kubectl debug 옵션으로 debug pod 띄우기

**Reply by t4e1** [reply]
kubectl debug -ti --image ubuntu --share-processes -n loki loki-backend-1

**Comment by keyolk** [comment]
go-rod 입니당

**Reply by t4e1** [reply]
수정했습니다 감사합니다 !
